examples_notebooks/community_contrib/couchbase/couchbasedb_demo.py

```diff
@@ -0,0 +1,264 @@
+)
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+INPUT_DIR = os.getenv("INPUT_DIR")
+COUCHBASE_CONNECTION_STRING = os.getenv("COUCHBASE_CONNECTION_STRING", "couchbase://localhost")
+COUCHBASE_USERNAME = os.getenv("COUCHBASE_USERNAME", "Administrator")
+COUCHBASE_PASSWORD = os.getenv("COUCHBASE_PASSWORD", "password")
+COUCHBASE_BUCKET_NAME = os.getenv("COUCHBASE_BUCKET_NAME", "graphrag-demo")
+COUCHBASE_SCOPE_NAME = os.getenv("COUCHBASE_SCOPE_NAME", "shared")
+COUCHBASE_VECTOR_INDEX_NAME = os.getenv("COUCHBASE_VECTOR_INDEX_NAME", "graphrag_index")
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+LLM_MODEL = os.getenv("LLM_MODEL", "gpt-4o")
+EMBEDDING_MODEL = os.getenv("EMBEDDING_MODEL", "text-embedding-ada-002")
+
+# Constants
+COMMUNITY_LEVEL = 2
+
+# Table names
+TABLE_NAMES = {
+    "COMMUNITY_REPORT_TABLE": "create_final_community_reports",
+    "ENTITY_TABLE": "create_final_nodes",
+    "ENTITY_EMBEDDING_TABLE": "create_final_entities",
+    "RELATIONSHIP_TABLE": "create_final_relationships",
+    "COVARIATE_TABLE": "create_final_covariates",
+    "TEXT_UNIT_TABLE": "create_final_text_units",
+}
+
+
+def load_data() -> dict[str, Any]:
+    """Load data from parquet files."""
+    logger.info("Loading data from parquet files")
+    data = {}
+
+    def load_table(table_name: str, read_function: Callable, *args) -> Any:
+        try:
+            table_data = pd.read_parquet(
+                f"{INPUT_DIR}/{TABLE_NAMES[table_name]}.parquet"
+            )
+            return read_function(table_data, *args)
+        except FileNotFoundError:
+            logger.warning(
+                "%(table_name)s file not found. Setting %(table_name_lower)s to None.",
+                {"table_name": table_name, "table_name_lower": table_name.lower()}
+            )
+            return None
+
+    data["entities"] = load_table("ENTITY_TABLE", read_indexer_entities, pd.read_parquet(f"{INPUT_DIR}/{TABLE_NAMES['ENTITY_EMBEDDING_TABLE']}.parquet"), COMMUNITY_LEVEL)
+    data["relationships"] = load_table("RELATIONSHIP_TABLE", read_indexer_relationships)
+    data["covariates"] = load_table("COVARIATE_TABLE", read_indexer_covariates)
+    data["reports"] = load_table("COMMUNITY_REPORT_TABLE", read_indexer_reports, pd.read_parquet(f"{INPUT_DIR}/{TABLE_NAMES['ENTITY_TABLE']}.parquet"), COMMUNITY_LEVEL)
+    data["text_units"] = load_table("TEXT_UNIT_TABLE", read_indexer_text_units)
+
+    logger.info("Data loading completed")
+    return data
+
+
+def setup_vector_store() -> CouchbaseVectorStore:
+    """Set up and connect to CouchbaseVectorStore."""
+    logger.info("Setting up CouchbaseVectorStore")
+    try:
+        description_embedding_store = CouchbaseVectorStore(
+            collection_name="entity_description_embeddings",
+            bucket_name=COUCHBASE_BUCKET_NAME,
+            scope_name=COUCHBASE_SCOPE_NAME,
+            index_name=COUCHBASE_VECTOR_INDEX_NAME,
+        )
+        description_embedding_store.connect(
```

Comment:
why we need a additional connect here, can't be done at CouchbaseVectorStore init